src/shader_recompiler/info.h

```diff
@@ -73,9 +73,44 @@ struct ImageResource {
     bool is_read{};
     bool is_written{};
 
+    [[nodiscard]] AmdGpu::ImageType GetBoundType(const AmdGpu::Image& image) const noexcept {
```

Comment:
Maybe these functions are better placed in AmdGpu::Image and pass is_array as a bool argument?